shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/cluster/coordinator/registry/cache/subscriber/ScalingRegistrySubscriber.java

```diff
@@ -130,6 +127,20 @@ public void clusterSwitchConfiguration(final ClusterSwitchConfigurationEvent eve
         persistService.persist(schemaName, event.getTargetRuleConfigs());
     }
     
+    /**
+     * scaling release lock.
+     *
+     * @param event Scaling Job Release Schema Name Lock Event
+     */
+    @Subscribe
+    public void scalingJobReleaseLockEvent(final ScalingJobReleaseSchemaNameLockEvent event) {
+        if (schemaNameLockedMap.getOrDefault(event.getSchemaName(), false)) {
+            log.info("scaling job finished, release schema name lock, event = {}", event);
+            lockRegistryService.releaseLock(decorateLockName(event.getSchemaName()));
+            schemaNameLockedMap.remove(event.getSchemaName());
+        }
+    }
```

Comment:
Is new event necessary for success case? Could we just do it in `scalingTaskFinished` like before, and add new logic for failure case later?

Reply:
ok, I will implement this feature in a new pull request